client/my-sites/plugins/plugin-meta/index.jsx

```diff
@@ -170,7 +170,7 @@ export default React.createClass( {
 				text={ i18n.translate( 'A new version is available.' ) }
 				status="is-warning"
 				button={ i18n.translate( 'Update to %(newPluginVersion)s', { args: { newPluginVersion: newVersion } } ) }
-				onClick={ this.handlePluginUpdates }
+				onDismissClick={ this.handlePluginUpdates }
```

Comment:
@johnHackworth yeah, current `onClick` is attached to the dismiss element, so it's misleading. Where is it supposed to work for the above case?